qiita_pet/handlers/study_handlers/sample_template.py

```diff
@@ -303,9 +312,19 @@ def patch(self):
         req_path = self.get_argument('path')
         req_value = self.get_argument('value', None)
         req_from = self.get_argument('from', None)
+        direct_upload = self.get_argument('direct_upload', False)
+
+        if direct_upload and direct_upload == 'true':
+            direct_upload = True
+
+            fd, req_value = mkstemp(suffix='.txt')
```

Comment:
The pattern here is unexpected, why get a handle and then close it? What about:

```python
with tempfile.NamedTemporaryFile(suffix='.txt') as fp:
     fp.write(self.request.files['value'][0]['body'])
```

This approach should not require the decode as the file is opened in binary.

Reply:
Good to know, will replace; thanks.